packer/plugin-getter/plugins.go

```diff
@@ -346,41 +382,63 @@ func (gp *GetOptions) ExpectedZipFilename() string {
 	return gp.expectedZipFilename
 }
 
-func (binOpts *BinaryInstallationOptions) CheckProtocolVersion(remoteProt string) error {
-	remoteProt = strings.TrimPrefix(remoteProt, "x")
-	parts := strings.Split(remoteProt, ".")
+type APIVersion struct {
+	Major int
+	Minor int
+}
+
+func NewAPIVersion(apiVersion string) (APIVersion, error) {
+	ver := APIVersion{}
+
+	apiVersion = strings.TrimPrefix(strings.TrimSpace(apiVersion), "x")
+	parts := strings.Split(apiVersion, ".")
 	if len(parts) < 2 {
-		return fmt.Errorf("Invalid remote protocol: %q, expected something like '%s.%s'", remoteProt, binOpts.APIVersionMajor, binOpts.APIVersionMinor)
+		return ver, fmt.Errorf(
+			"Invalid remote protocol: %q, expected something like '%s.%s'",
+			apiVersion, pluginsdk.APIVersionMajor, pluginsdk.APIVersionMinor,
+		)
 	}
-	vMajor, vMinor := parts[0], parts[1]
 
-	// no protocol version check
-	if binOpts.APIVersionMajor == "" && binOpts.APIVersionMinor == "" {
-		return nil
+	vMajor, err := strconv.Atoi(parts[0])
+	if err != nil {
+		return ver, err
 	}
+	ver.Major = vMajor
 
-	if vMajor != binOpts.APIVersionMajor {
-		return fmt.Errorf("Unsupported remote protocol MAJOR version %q. The current MAJOR protocol version is %q."+
-			" This version of Packer can only communicate with plugins using that version.", vMajor, binOpts.APIVersionMajor)
+	vMinor, err := strconv.Atoi(parts[1])
+	if err != nil {
+		return ver, err
 	}
+	ver.Minor = vMinor
+
+	return ver, nil
+}
 
-	if vMinor == binOpts.APIVersionMinor {
+func (binOpts *BinaryInstallationOptions) CheckProtocolVersion(remoteProt string) error {
+	// no protocol version check
+	if binOpts.APIVersionMajor == "" && binOpts.APIVersionMinor == "" {
 		return nil
 	}
 
-	vMinori, err := strconv.Atoi(vMinor)
+	// Should never error if both components are set
+	localVersion, err := NewAPIVersion(fmt.Sprintf("x%s.%s", binOpts.APIVersionMajor, binOpts.APIVersionMinor))
 	if err != nil {
-		return err
+		panic(fmt.Sprintf("malformed API version in Packer. This is a programming error, please open an error to report it."))
```

Comment:
Why do we panic here instead of returning an error?

Reply:
Ah good question; this parsing is done on static data that is normally always valid: the API versions embedded in Packer (i.e. those from the SDK).
If it goes wrong, we are in an invalid state, and we definitely don't want to continue operating, since we'll never be able to check compatibility with plugins, protocol-wise at least.
Come to think of it, this may become an `init` time check, we probably want to crash ASAP if that somehow breaks someday.
I'll make the change before merging this PR